### PR TITLE
Initial Prototype of Novice User Portal

### PR DESCRIPTION
**Pull request by jpwhite4**
Initial version of the novice user tab. I expect there will be some additional changes needed once we get the various portlets fully integrated. There will also be further updates to the documentation.

The main difference between this and the previous novice_user feature branch is that the novice user tab is now enabled/disabled by the portal_settings.ini setting and you have to add your portlet javascript file to the assets.json (or assets.d/MODULE.json for modules).

### CODE REVIEW

File: docs/novice_user.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        }
+    }
+});
+
+/**
+ * The Ext.reg call is used to register an xtype for this class so it
+ * can be dynamically instantiated
+ */
+Ext.reg('DeveloperPortlet', XDMoD.Modules.SummaryPortlets.DeveloperPortlet);
+```
```
Comment (ryanrath) [inline docs/novice_user.md]: Would examples of the additions required for `etc/assets.json` and `etc/roles.d/summary.json` be useful too?

File: docs/novice_user.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+         * in the top toolbar are changed by the user or when the refresh
+         * button is clicked.
+         *
+         * A typical portlet will reload its content with the updated
+         * duration parameters.
+         */
+        duration_change: function (timeframe) {
+            var console = this.getComponent('console');
+            console.body.insertHtml('beforeBegin', '<pre>' + JSON.stringify(timeframe, null, 4) + '</pre>');
+        }
```
Comment (mkzia) [inline docs/novice_user.md]: You are missing a comma!

**Comment by mkzia** [comment]
I have tested the new method and it works fine for me. My branch is https://github.com/mkzia/xdmod/tree/mkzia_novice

**Comment by ryanrath** [comment]
I also have things working w/ this branch + my two novice user components: 
  - Summary Statistics: https://github.com/ryanrath/xdmod/tree/novice_summary_statistics\
  - Center Report Card: https://github.com/ryanrath/xdmod/tree/novice_appkernels
    - App Kernels: https://github.com/ryanrath/xdmod-appkernels/tree/novice_appkernels
    - CCR Private 

I have the Center Report Card component up on my dev port. At the moment it's only enabled for Center Director / Staff.

**Pull request merged**